proposals/0000-http-pinning-service-api.md

```diff
@@ -0,0 +1,216 @@
+API, and the Dat Pinning API. These resources should be indicated by the
+`user-accounts-api.com/v1` and `datprotocol.com/deps/0000-http-pinning-service-api`
+rel-types, respectively. (These rel-types will need to be updated
+with the final URLs for their specifications.) If either API is absent from
+the PSA document, the service will be rejected.
+
+
+# User accounts API
+[user-accounts-api]: #user-accounts-api
+
+The user-accounts API should provide the following resources:
+
+```
+POST /register  Register a new account.
+POST /verify    Verify the email address of a recently-registered account.
+POST /login     Create a new session with an existing account.
+POST /logout    End a session.
+GET  /account   Get information about the account attached to the current session.
+POST /account   Update information about the account attached to the current session.
+```
+
+TODO- specify each route in detail.
+
+Full documentation for this API should be made available at https://user-accounts-api.com.
+
+## User registration flow
+[user-registration-flow]: #user-registration-flow
+
+### Step 1. Register
+
+User POSTS to `/register` with body:
+
+```
+{
+  email: String
+  username: String
+  password: String
+}
+```
+
+Server creates a new account for the user. A random 32-byte email-verification
+nonce is created. The user record indicates:
+
+scopes|isEmailVerified|emailVerifyNonce
+------|---------------|----------------
+none|false|XXX
+
+Server sends an email to the user with the `emailVerifyNonce`. 
+
+Server responds 200 or 204.
+
+### Step 2. Verify (POST /verify)
+
+User POSTS `/v1/verify` with JSON body:
+
+```
+{
+  username: String, username of the account
+  nonce: String, verification nonce
+}
+```
+
+Server updates user record to indicate:
+
+scopes|isEmailVerified|emailVerifyNonce
+------|---------------|----------------
+user|true|null
+
+Sever generates a session and session token, and responds 200 with a JSON body:
+
+```
+{
+  sessionToken: String, users session token
+}
+```
+
+## Session login flow
+[session-login-flow]: #session-login-flow
+
+User POSTS to `/login` with body:
+
+```
+{
+  username: String
+  password: String
+}
+```
+
+Sever generates a session and session token, and responds 200 with a JSON body:
+
+```
+{
+  sessionToken: String, users session token
+}
+```
+
+
+# Dat pinning API
+[dat-pinning-api]: #dat-pinning-api
+
+The dat pinning API should provide the following resources:
+
+```
+GET  /            List all Dat data pinned by this account.
+POST /add         Add a Dat to this account's list of pins.
+POST /remove      Remove a Dat from this account's list of pins.
+GET  /item/:key   Get information about a Dat in the account's list of pins.
+                  Key may be the pubkey or name of the dat.
+POST /item/:key   Update information about a Dat in the account's list of pins.
+                  Key may be the pubkey or name of the dat.
```

Comment:
do we need to discuss the payloads of the above or just make it up as we go along?